gobblin-metastore/src/main/resources/gobblin_job_history_store.sql

```diff
@@ -91,29 +91,29 @@ CREATE TABLE IF NOT EXISTS gobblin_task_metrics (
 );
 
 CREATE TABLE IF NOT EXISTS gobblin_job_properties (
-  property_id BIGINT(21) NOT NULL AUTO_INCREMENT,
-  job_id VARCHAR(128) NOT NULL,
-  property_key VARCHAR(128) NOT NULL,
-  property_value TEXT NOT NULL,
+	property_id BIGINT(21) NOT NULL AUTO_INCREMENT,
+	job_id VARCHAR(128) NOT NULL,
+	property_key VARCHAR(128) NOT NULL,
+	property_value MEDIUMTEXT NOT NULL,
```

Comment:
AFAIK, this is not how the text column works.  In general, all flavors of the TEXT column (TINYTEXT, TEXT, MEDIUMTEXT, and LONGTEXT) store the data outside the row.  Additionally, they operate similar to VARCHAR columns in that they only use the amount of space needed to store the data plus some additional metadata.  Since the TEXT data is stored off row, there is some additional cost when reading the data.

On another note, because of how the current state system works, all job properties are duplicated on the tasks.  Task may then contain additional properties (or overrides).  I think it would be worthwhile to de-dup the task properties against the job properties when storing the data.  The down side is that any consumer of the history would need to know that the tasks have all the job properties.  The advantage is that it would greatly decrease the amount of data stored in the DB and returned by the REST service.  What do you think?